src/core/dep-graph.ts

```diff
@@ -127,9 +130,11 @@ class DepGraphImpl implements types.DepGraphInternal {
   public pkgPathsToRoot(pkg: types.Pkg): types.PkgInfo[][] {
     const pathsToRoot: types.PkgInfo[][] = [];
     for (const nodeId of this.getPkgNodeIds(pkg)) {
-      const pathsFromNodeToRoot = this.pathsFromNodeToRoot(nodeId);
+      const [pathsFromNodeToRoot] = this.pathsFromNodeToRoot(nodeId);
       for (const path of pathsFromNodeToRoot) {
-        pathsToRoot.push(path);
+        // the reason for the .slice() is to create a new path array for every call.
```

Comment:
You've added an unrelated functional change without adding tests

Reply:
Actually that's probably not true. Previously "paths" would be generated from scratch.. and so external mutations wouldn't matter. Now that we're caching things we can't let external callers mutate the data in the cache.